### DISCUSSION TITLE
Shut down jupyter server from VSCode

### DISCUSSION THREAD

**Discussion by PSAB** [microsoft/vscode-jupyter#15149] [Questions and Answers]
I'm running a jupyter notebook for ML locally on my machine (mac w/ intel) in a conda environment. There doesn't seem to be a nice way to shut down the Jupyter Server from VSCode, unlike running Jupyter notebooks in the browser where you can shut down a server whenever desired. 

So even when I restart the kernel and quit VSCode, there's a Python process that gets stuck on my machine, and I have to manually kill it from terminal or activity monitor. This seems too trivial, is there a way to shut down a jupyter server from VSCode that I'm missing?

**Comment by DonJayamanne** [comment]
> So even when I restart the kernel and quit VSCode, there's a Python process that gets stuck on my machine, 

Closing the notebook or VS Code should shutdown the kernel.
Please can you do the following:
* Open settings & go into `Jupyter->Logging`
* Change the value to `verbose`
* Reload VS Code,
* Close all notebooks
* Open the Jupyter output panel using the command `Jupyter: Show Output` and clear the output
* Run one cell against the kernel you ran into issues, ensure the code you run is very simple such as `print("Hello World")`
* Next, restart the kernel as you described
* Close the notebook
* See if the process is still running, if its still running, then please copy all of the logs and past it into a github issue.

**Reply by PSAB** [reply]
Hi @DonJayamanne It seems to work fine when I follow your steps. According to the Jupyter output panel, the kernel is disposed when I close the notebook. I wish there was a way to shutdown a kernel without having to close the notebook.

**Reply by DonJayamanne** [reply]
You can try Jupyter Powertools extension,
